spec/models/generic_file_spec.rb

```diff
@@ -625,4 +625,156 @@
+
+      allow(gv_black_photo_sub_collection_double).to receive(:map).and_return([GenericFile::GV_BLACK_PHOTOGRAPH_SUB_COLLECTION_ID])
+      allow(gv_black_photo_sub_collection_double).to receive(:id)
+
+      allow(gv_black_collection_double).to receive(:map).and_return([GenericFile::GV_BLACK_COLLECTION_ID])
+      allow(gv_black_collection_double).to receive(:id)
+    end
+
+    context "record is public" do
+      before do
+        # GenericFile validates the type of permissions so simply passing a double to the constructor does not work
+        public_record.stub(:permissions).and_return([public_permission_double])
+      end
+
+      context "has doi" do
+        before do
+          public_record.stub(:doi).and_return(random_doi)
+        end
+
+        context "is in the gv black photograph sub collection" do
+          context "is NOT in the gv black papers collection" do
+            before do
+              public_record.stub(:collections).and_return([gv_black_photo_sub_collection_double])
+            end
+
+            it "returns true" do
+              expect(public_record.unexportable?).to eq(true)
+            end
+          end
+
+          context "is in the gv black papers collection" do
+            before do
+              public_record.stub(:collections).and_return([gv_black_photo_sub_collection_double, gv_black_collection_double])
+            end
+
+            it "returns true" do
+              expect(public_record.unexportable?).to eq(true)
+            end
+          end
+        end
+
+        context "is NOT in the gv black photograph sub collection" do
+          context "is NOT in the gv black papers collection" do
+            before do
+              public_record.stub(:collections).and_return([])
+            end
+
+            it "returns true" do
+              expect(public_record.unexportable?).to eq(true)
+            end
+          end
+
+          context "is in the gv black papers collection" do
+            before do
+              public_record.stub(:collections).and_return([gv_black_collection_double])
+            end
+
+            it "returns true" do
+              expect(public_record.unexportable?).to eq(true)
+            end
+          end
+        end
+      end
+
+      context "has no doi" do
+        before do
+          public_record.stub(:doi).and_return(blank_doi)
+        end
+
+        context "is in the gv black photograph sub collection" do
+          before do
+            public_record.stub(:collections).and_return(gv_black_photo_sub_collection_double)
+          end
+
+          context "is NOT in the gv black papers collection" do
+            it "returns true" do
+              expect(public_record.unexportable?).to eq(false)
+            end
+          end
+        end
+      end
+    end
+
+    context "record is NOT public" do
+      before do
+        allow(private_record).to receive(:permissions).and_return([])
```

Comment:
I'd be willing to change `public/private_record` to just `record` since within individual context blocks we can stub out methods to have different return values. I thought that it was helpful to clarify what was going on by having two separate objects, but it does mean an extra database hit.